--- a/src/base/os_linux/os_system_linux.c
+++ b/src/base/os_linux/os_system_linux.c
@@ -93,7 +93,15 @@ static void _flush_netlink_buffer(struct
 static struct sockaddr_nl _netlink_nladdr = { .nl_family = AF_NETLINK };
 
 static struct iovec _netlink_rcv_iov;
-static struct msghdr _netlink_rcv_msg = { &_netlink_nladdr, sizeof(_netlink_nladdr), &_netlink_rcv_iov, 1, NULL, 0, 0 };
+static struct msghdr _netlink_rcv_msg = {
+  .msg_name = &_netlink_nladdr,
+  .msg_namelen = sizeof(_netlink_nladdr),
+  .msg_iov = &_netlink_rcv_iov,
+  .msg_iovlen = 1,
+  .msg_control = NULL,
+  .msg_controllen = 0,
+  .msg_flags = 0,
+};
 
 static struct nlmsghdr _netlink_hdr_done = { .nlmsg_len = sizeof(struct nlmsghdr), .nlmsg_type = NLMSG_DONE };
 
@@ -102,8 +110,15 @@ static struct iovec _netlink_send_iov[]
   { &_netlink_hdr_done, sizeof(_netlink_hdr_done) },
 };
 
-static struct msghdr _netlink_send_msg = { &_netlink_nladdr, sizeof(_netlink_nladdr), _netlink_send_iov,
-  ARRAYSIZE(_netlink_send_iov), NULL, 0, 0 };
+static struct msghdr _netlink_send_msg = {
+  .msg_name = &_netlink_nladdr,
+  .msg_namelen = sizeof(_netlink_nladdr),
+  .msg_iov = _netlink_send_iov,
+  .msg_iovlen = ARRAYSIZE(_netlink_send_iov),
+  .msg_control = NULL,
+  .msg_controllen = 0,
+  .msg_flags = 0,
+};
 
 /* netlink timeout handling */
 static struct oonf_timer_class _netlink_timer = {
